staticaddr/loopin/fsm.go

```diff
@@ -0,0 +1,383 @@
+package loopin
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2/schnorr/musig2"
+	"github.com/lightninglabs/loop/fsm"
+	"github.com/lightninglabs/loop/staticaddr/deposit"
+	"github.com/lightninglabs/loop/staticaddr/version"
+)
+
+// FSM embeds an FSM and extends it with a static address loop-in and a config.
+type FSM struct {
+	*fsm.StateMachine
+
+	cfg *Config
+
+	// loopIn stores the loop-in details that are relevant during the
+	// lifetime of the swap.
+	loopIn *StaticAddressLoopIn
+
+	// MuSig2 data must not be re-used across restarts, hence it is not
+	// persisted.
+	//
+	// htlcServerNonces contains all the nonces that the server generated
+	// for the htlc musig2 sessions.
```

Comment:
Should the two parts of the comment be swapped?

Reply:
I am unsure what you mean.